packages/dids/src/methods/did-dht.ts

```diff
@@ -0,0 +1,1550 @@
+      throw new DidError(DidErrorCode.InvalidDidDocumentLength, `Pkarr response must be at least 72 bytes but got: ${messageBytes.byteLength}`);
+    }
+
+    if (messageBytes.byteLength > 1072) {
+      throw new DidError(DidErrorCode.InvalidDidDocumentLength, `Pkarr response exceeds 1000 byte limit: ${messageBytes.byteLength}`);
+    }
+
+    // Decode the BEP44 message from the byte array.
+    const bep44Message: Bep44Message = {
+      k   : publicKeyBytes,
+      seq : Number(new DataView(messageBytes).getBigUint64(64)),
+      sig : new Uint8Array(messageBytes, 0, 64),
+      v   : new Uint8Array(messageBytes, 72)
+    };
+
+    return bep44Message;
+  }
+
+  /**
+   * Publishes a signed BEP44 message to a Pkarr relay server.
+   *
+   * @see {@link https://github.com/Nuhvi/pkarr/blob/main/design/relays.md | Pkarr Relay design}
+   *
+   * @param relay - The Pkarr relay server URL.
+   * @param bep44Message - The BEP44 message to be published, containing the signed DNS packet.
+   * @returns A promise resolving to `true` if the message was successfully published, otherwise `false`.
+   */
+  private static async pkarrPut({ relay, bep44Message }: {
+    bep44Message: Bep44Message;
+    relay: string;
+  }): Promise<boolean> {
+    // The identifier (key in the DHT) is the z-base-32 encoding of the Identity Key.
+    const identifier = Convert.uint8Array(bep44Message.k).toBase32Z();
+
+    // Concatenate the Pkarr relay URL with the identifier to form the full URL.
+    const url = new URL(identifier, relay).href;
+
+    // Construct the body of the request according to the Pkarr relay specification.
+    const body = new Uint8Array(bep44Message.v.length + 72);
+    body.set(bep44Message.sig, 0);
+    new DataView(body.buffer).setBigUint64(bep44Message.sig.length, BigInt(bep44Message.seq));
+    body.set(bep44Message.v, bep44Message.sig.length + 8);
+
+    // Transmit the Put request to the Pkarr relay and get the response.
+    let response: Response;
+    try {
+      response = await fetch(url, {
+        method  : 'PUT',
+        headers : { 'Content-Type': 'application/octet-stream' },
+        body
+      });
+
+    } catch (error: any) {
+      throw new DidError(DidErrorCode.InternalError, `Failed to put Pkarr record: ${error.message}`);
+    }
+
+    // Return `true` if the DHT request was successful, otherwise return `false`.
+    return response.ok;
+  }
+
+  /**
+   * Converts a DNS packet to a DID document according to the DID DHT specification.
+   *
+   * @see {@link https://did-dht.com/#dids-as-dns-records | DID DHT Specification, § DIDs as DNS Records}
+   *
+   * @param params - The parameters to use when converting a DNS packet to a DID document.
+   * @param params.didUri - The DID URI of the DID document.
+   * @param params.dnsPacket - The DNS packet to convert to a DID document.
+   * @returns A promise that resolves to a DID document.
+   */
+  private static async fromDnsPacket({ didUri, dnsPacket }: {
```

Comment:
this is pretty awesome! There is not lib for this? 

If not we can eventually extract this out and make our own lib

Reply:
There's a DNS packet lib that we use for encoding to DNS wire format.  It is used by our DID DHT implementation.  The `fromDnsPacket()` and `toDnsPacket()` implementations in `did-dht.ts` are specific to the encoding/decoding of DID DHT documents.